apis/common/v1/condition.go

```diff
@@ -101,7 +101,8 @@ func (c Condition) Equal(other Condition) bool {
 	return c.Type == other.Type &&
 		c.Status == other.Status &&
 		c.Reason == other.Reason &&
-		c.Message == other.Message
+		c.Message == other.Message &&
+		c.ObservedGeneration == other.ObservedGeneration
```

Comment:
Updating to not treat ObservedGeneration as something to ignore between versions of conditions.

Reply:
Looks like the godoc string needs an update for this too.